### PR TITLE
cjson: init at 1.7.14

### LINKED ISSUE #107161: cJSON
**Project description**
cJSON aims to be the dumbest possible parser that you 
can get your job done with. It's a single file of C, 
and a single header file.

- This is a new [build requirement](https://github.com/eclipse/mosquitto#build-dependencies) for mosquitto > 2.0.0. 
- There is a possible name clash as there is already a Lua and a Python implementation which are using `cjson` as part of their name.

**Metadata**

* homepage URL: https://github.com/DaveGamble/cJSON
* source URL: https://github.com/DaveGamble/cJSON/archive/v1.7.14.tar.gz
* license: MIT and ASL 2.0
* platforms: Unix, unsure about the the others

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Motivation for this change

Closes #107161

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

CC @fabaff for testing

### CODE REVIEW

File: pkgs/development/libraries/cjson/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    rev = "v${version}";
+    sha256 = "1a3i9ydl65dgwgmlg79n5q8qilmjkaakq56sam1w25zcrd8jy11q";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/DaveGamble/cJSON";
+    description = "Ultralightweight JSON parser in ANSI C ";
+    license = licenses.mit;
```
Comment (fabaff) [inline pkgs/development/libraries/cjson/default.nix:19]: A part of the test suite is licensed under [ASL 2.0](https://github.com/DaveGamble/cJSON/blob/master/tests/json-patch-tests/README.md#license).

File: pkgs/development/libraries/cjson/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    repo = "cJSON";
+    rev = "v${version}";
+    sha256 = "1a3i9ydl65dgwgmlg79n5q8qilmjkaakq56sam1w25zcrd8jy11q";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/DaveGamble/cJSON";
+    description = "Ultralightweight JSON parser in ANSI C ";
```
Comment (fabaff) [inline pkgs/development/libraries/cjson/default.nix:18]:
```suggestion
    description = "Ultralightweight JSON parser in ANSI C";
```

File: pkgs/development/libraries/cjson/default.nix (hunk 1 above)
Comment (matthiasbeyer) [reply pkgs/development/libraries/cjson/default.nix:19]: How to have multiple licenses in a package in nixpkgs?

Comment (Emantor) [reply pkgs/development/libraries/cjson/default.nix:19]:
```
license = with licenses; [ mit asl20 ];
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 107169` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cjson</li>
  </ul>
</details>

**Comment by matthiasbeyer** [comment]
squashed.

File: pkgs/development/libraries/cjson/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchFromGitHub {
+    owner = "DaveGamble";
+    repo = "cJSON";
+    rev = "v${version}";
+    sha256 = "1a3i9ydl65dgwgmlg79n5q8qilmjkaakq56sam1w25zcrd8jy11q";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cjson/default.nix:16]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/cjson/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{stdenv, fetchFromGitHub, cmake}:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cjson/default.nix:1]:
```suggestion
{ lib, fetchFromGitHub, cmake }:
```

Comment (Emantor) [reply pkgs/development/libraries/cjson/default.nix]: How does this work if the derivation uses `stdenv.mkDerivation`?

File: pkgs/development/libraries/cjson/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{ lib, fetchFromGitHub, cmake }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cjson/default.nix:1]:
```suggestion
{ stdenv, lib, fetchFromGitHub, cmake }:
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107169` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>cjson: log <a href="https://termbin.com/y2ty">https://termbin.com/y2ty</a></li>
  </ul>
</details>

File: pkgs/development/libraries/cjson/default.nix (hunk 4 above)
Comment (prusnak) [inline pkgs/development/libraries/cjson/default.nix:1]: You need to include both `stdenv` and `lib`. Add space after `{` and before `}`. 

```suggestion
{ stdenv, lib, fetchFromGitHub, cmake }:
```

File: pkgs/development/libraries/cjson/default.nix [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    rev = "v${version}";
+    sha256 = "1a3i9ydl65dgwgmlg79n5q8qilmjkaakq56sam1w25zcrd8jy11q";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  meta = with lib; {
+    homepage = "https://github.com/DaveGamble/cJSON";
+    description = "Ultralightweight JSON parser in ANSI C";
+    license = with licenses; [ mit asl20 ];
```
Comment (prusnak) [inline pkgs/development/libraries/cjson/default.nix:19]: cJSON licence is MIT

```suggestion
    license = licenses.mit;
```

**Review by prusnak — CHANGES_REQUESTED** [review]

**Comment by matthiasbeyer** [comment]
Updated.

**Comment by prusnak** [comment]
Please squash the fixup commit

**Comment by matthiasbeyer** [comment]
On 01-02-2021 09:07:28, Pavol Rusnak wrote:
> Please squash the fixup commit

Done.

**Pull request merged**